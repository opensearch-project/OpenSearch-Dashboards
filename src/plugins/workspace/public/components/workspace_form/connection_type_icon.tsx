/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import React from 'react';
import { EuiIcon } from '@elastic/eui';

import prometheusLogo from '../../assets/prometheus_logo.svg';
import s3Logo from '../../assets/s3_logo.svg';
import cloudWatchLogo from '../../assets/cloudwatch_logo.svg';
import securityLakeLogo from '../../assets/security_lake_logo.svg';
import { DataConnectionType } from '../../../../data_source/common/';

// Direct query connection and data connection both have different types, each type has a corresponding icon
export const ConnectionTypeIcon = ({ type }: { type?: string }) => {
  switch (type) {
    case 'Amazon S3':
      return <EuiIcon type={s3Logo} />;
    case 'Prometheus':
      return <EuiIcon type={prometheusLogo} />;
    case DataConnectionType.CloudWatch:
      return <EuiIcon type={cloudWatchLogo} />;
    case DataConnectionType.SecurityLake:
      return <EuiIcon type={securityLakeLogo} />;
    default:
      return null;
  }
};
