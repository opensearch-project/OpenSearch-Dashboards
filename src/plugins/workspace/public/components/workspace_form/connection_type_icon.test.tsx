/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import React from 'react';
import { mount } from 'enzyme';

import { ConnectionTypeIcon } from './connection_type_icon';

describe('ConnectionTypeIcon', () => {
  it('should render normally', () => {
    expect(mount(<ConnectionTypeIcon />)).toMatchSnapshot();
    expect(mount(<ConnectionTypeIcon type="Amazon S3" />)).toMatchSnapshot();
  });
});
