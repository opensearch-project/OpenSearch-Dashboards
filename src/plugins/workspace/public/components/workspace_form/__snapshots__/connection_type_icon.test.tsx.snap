// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConnectionTypeIcon should render normally 1`] = `<ConnectionTypeIcon />`;

exports[`ConnectionTypeIcon should render normally 2`] = `
<ConnectionTypeIcon
  type="Amazon S3"
>
  <EuiIcon
    type="test-file-stub"
  >
    <span
      data-euiicon-type="test-file-stub"
    />
  </EuiIcon>
</ConnectionTypeIcon>
`;
